Sources/PKYouboraAdsAdapter.swift

```diff
@@ -240,6 +278,36 @@ extension PKYouboraAdsAdapter {
                         self.fireFatalError(withMessage: adErrorEvent.localizedDescription, code: "\(adErrorEvent.code)", andMetadata: adErrorEvent.description)
                     }
                 }
+            case let e where e.self == AdEvent.adFirstQuartile:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    guard let self = self else { return }
+                    
+                    self.fireQuartile(1)
+                }
+            case let e where e.self == AdEvent.adMidpoint:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    guard let self = self else { return }
+                    
+                    self.fireQuartile(2)
+                }
+            case let e where e.self == AdEvent.adThirdQuartile:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    guard let self = self else { return }
+                    
+                    self.fireQuartile(3)
+                }
+            case let e where e.self == AdEvent.adBreakStarted:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    guard let self = self else { return }
+                    
+                    self.fireAdBreakStart()
+                }
+            case let e where e.self == AdEvent.adBreakEnded:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    guard let self = self else { return }
+                    
+                    self.fireAdBreakStop()
```

Comment:
we only fire these events here - I think the other places with the try logic are redundant.
The event model should be predictable.

Reply:
Maybe is easier to try to explain here why the `adBreaks` are not only here and we have the `tryAdBreak` methods as well. This `AdEvent.adBreakStarted` and `AdEvent.adBreakEnded` are only available for IMA DAI. But for instance in IMA you can have as well an ad break with more then 1 pod and we consider that an ad break as well. So the workaround that we found to catch the adBreaks in IMA is check the pods and fire the `adBreakStart` when the first ad of the pod will be started and the `fireAdBreakStop` when the last pod of this ad interval ends. That's the reason why every time an ad starts or ends we check if if the first or the last pod and we trigger the event. This way we can have adBreaks working with IMA and IMA DAI.